client/src/main/java/io/split/client/RequestDecorator.java

```diff
@@ -0,0 +1,60 @@
+package io.split.client;
+
+import org.apache.hc.client5.http.classic.methods.HttpUriRequestBase;
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Arrays;
+import java.util.Set;
+
+class NoOpHeaderDecorator implements  UserCustomHeaderDecorator {
+    public NoOpHeaderDecorator() {}
+    @Override
+    public Map<String, String> getHeaderOverrides() {
+        return new HashMap<>();
+    }
+}
+
+public final class RequestDecorator {
+    UserCustomHeaderDecorator _headerDecorator;
+
+    private static final Set<String> forbiddenHeaders = new HashSet<>(Arrays.asList(
+            "splitsdkversion",
+            "splitmachineip",
+            "splitmachinename",
+            "splitimpressionsmode",
+            "host",
+            "referrer",
+            "content-type",
+            "content-length",
+            "content-encoding",
+            "accept",
+            "keep-alive",
+            "x-fastly-debug"
+    ));
+
+    public RequestDecorator(UserCustomHeaderDecorator headerDecorator) {
+        _headerDecorator = (headerDecorator == null)
+                ? new NoOpHeaderDecorator()
+                : headerDecorator;
+    }
+
+    public HttpUriRequestBase decorateHeaders(HttpUriRequestBase request) {
+        try {
+            Map<String, String> headers = _headerDecorator.getHeaderOverrides();
+            for (Map.Entry entry : headers.entrySet()) {
+                if (isHeaderAllowed(entry.getKey().toString())) {
+                    request.addHeader(entry.getKey().toString(), entry.getValue());
```

Comment:
when using addHeader, there will be only one  Authorization header with multiple values separated by comma.  This does not impact the proxy auth since we support "Basic xxxxxx" which is different than the default "Bearer xxxxx"